prime-router/src/main/kotlin/azure/observability/event/AzureEventService.kt

```diff
@@ -30,13 +32,19 @@ class AzureEventServiceImpl(
         logger.debug("Sending event of type $name to Azure AppInsights")
         telemetryClient.trackEvent(name, event.serialize(), emptyMap())
     }
+
+    override fun trackEvent(eventName: ReportStreamEventName, event: AzureCustomEvent) {
+        logger.debug("Sending event of type $eventName to Azure AppInsights")
+        telemetryClient.trackEvent(eventName.name, event.serialize(), emptyMap())
+    }
 }
 
 /**
  * Local storage of azure events (only used for testing)
  */
 class LocalAzureEventServiceImpl(
```

Comment:
This whole class is basically a carbon copy of the one I wrote in the test project (`InMemoryAzureEventService`). Maybe we should just move that one over here so we don't have multiple test implementations.

Reply:
I agree, though I'll open a tech debt ticket for that since this existed before this PR and the changeset is already very large